app/cdap/components/GridTable/components/DirectiveComponents/Calculate/index.tsx

```diff
@@ -0,0 +1,158 @@
+import { ICalculateProps } from './types';
+
+export default function({
+  functionName,
+  setDirectiveComponentsValue,
+  directiveComponentValues,
+}: ICalculateProps) {
+  const [customInput, setCustomInput] = useState<string>('');
+  const [copyToNewColumn, setCopyToNew] = useState<boolean>(false);
+  const [column, setColumnName] = useState<string>('');
+  const [isError, setIsError] = useState<boolean>(false);
+  const UI_INPUT =
+    CALCULATE_OPTIONS?.length > 0
+      ? CALCULATE_OPTIONS.filter((el) => el?.value === functionName)
+      : [];
+  const classes = useStyles();
+  useEffect(() => {
+    setDirectiveComponentsValue({ ...directiveComponentValues, customInput });
+  }, [customInput]);
+  useEffect(() => {
+    setDirectiveComponentsValue({ ...directiveComponentValues, copyToNewColumn });
+  }, [copyToNewColumn]);
+  useEffect(() => {
+    if (directiveComponentValues?.columnNames?.filter((el: string) => el === column).length) {
+      setIsError(true);
+    } else {
+      setIsError(false);
+    }
+    setDirectiveComponentsValue({ ...directiveComponentValues, copyColumnName: column });
+  }, [column]);
+
+  return (
+    <Box className={classes.calculateWrapper}>
+      {UI_INPUT.length > 0 &&
+        UI_INPUT.map((item) =>
+          item.value === 'CHARCOUNT' ? (
+            <Box className={classes.calculateFlex}>
+              <FormGroup>
+                <LabelComponent
+                  labelText={`${T.translate(
+                    'features.DirectiveUIComponent.calculate.nameNewColumn'
+                  )}`}
+                />
+                <FormInputFieldComponent
+                  formInputValue={column}
+                  classnames={classes.formFieldStyles}
+                  inputProps={{
+                    classes: { underline: classes.underlineStyles, input: classes.inputStyles },
+                    type: 'text',
+                    value: column,
+                    onChange: (e: React.ChangeEvent<HTMLInputElement>) =>
+                      setColumnName(e.target.value),
+                    color: 'primary',
+                    placeholder: `${T.translate(
+                      'features.DirectiveUIComponent.calculate.destinationColumn'
+                    )}`,
+                  }}
+                />
+                {isError && (
+                  <FormHelperText error={isError}>
+                    {T.translate('features.DirectiveUIComponent.calculate.columnExist')}
+                  </FormHelperText>
+                )}
+              </FormGroup>
+            </Box>
+          ) : item.inputRequired ? (
+            <Box className={classes.calculateFlex}>
+              {item?.sign && (
+                <Typography className={`${classes.formLabelStyles} ${classes.signText}`}>
+                  {item?.sign}
+                </Typography>
+              )}
+              <FormGroup>
+                <FormInputFieldComponent
```

Comment:
try not to call components multiple times, rather optimise code based on conditions and requirement

Reply:
Done